arrows/vtk/CMakeLists.txt

```diff
@@ -24,9 +28,11 @@ kwiver_add_library( kwiver_algo_vtk
   )
 
 target_link_libraries( kwiver_algo_vtk
-  PUBLIC               vital
+  PRIVATE              kwiver_algo_core
```

Comment:
`vital` should be PUBLIC.  `kwiver_algo_core` should not be included at all.

Reply:
kwiver_algo_core is needed otherwise:
<pre>
CMakeFiles/kwiver_algo_vtk.dir/mesh_coloration.cxx.o: In function `kwiver::arrows::vtk::mesh_coloration::mesh_coloration(std::shared_ptr<kwiver::vital::config_block>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<kwiver::vital::config_block>&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<kwiver::vital::camera_map>&)':
mesh_coloration.cxx:(.text+0x6e02): undefined reference to `kwiver::vital::algorithm_def<kwiver::vital::algo::video_input>::set_nested_algo_configuration(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<kwiver::vital::config_block>, std::shared_ptr<kwiver::vital::algo::video_input>&)'
</pre>

Reply:
@dstoup Should a module be public just because one of its headers is used in this module? I don't think it should. The user should specifically request the module that owns the header. So the VTK modules are added PRIVATE for now.

Reply:
@danlipsa try `vital_algo` instead of `kwiver_core_algo`

Reply:
> @dstoup Should a module be public just because one of its headers is used in this module? I don't think it should. The user should specifically request the module that owns the header. So the VTK modules are added PRIVATE for now.

Not exactly. A vtk module should be linked PUBLIC if one of it's headers is used in this library's header, because that is adding the vtk module to this libraries public API. If you used the vtk headers in the cxx file instead (or in a private header), then it can remain PRIVATE since you're not publicly exposing the vtk headers use to consumers. Keeping the links as PRIVATE will cause link errors when you create a new target that link to this one, unless the new target also links the publicly exposed vtk module, which is not something we should be pushing onto consumers.

*Note, some compilers are more forgiving of an incorrect link interface, but some, e.g. clang, are very strict about it.

Reply:
OK, It makes sense. vtkRenderingCore has vtkCamera in it, so I'll make that PUBLIC.

Reply:
@dstoup The problem is this library is not passed correctly to TeleSculptor. I get the following error:
 35%] Linking CXX executable ../bin/TeleSculptor
/usr/bin/x86_64-linux-gnu-ld: cannot find -lvtkRenderingCore

This is the reason I made those VTK libraries PRIVATE in the first place. TeleSculptor already links with VTK libraries. Any ideas?

Reply:
@dstoup I don't fully understand why including a header file from VTK would require making that VTK library public?
So vtkKwiverCamera.h includes vtkCamera.h Why does kwiver_algo_vtk has to link with vtkRenderingCore (which includes vtkCamera) PUBLIC? Linking with it PRIVATE would resolve all symbols referenced from vtkKwiverCamera. Unless vtkCamera references another static symbols (which is not the case) a client should not get unreferenced symbols.

Could we commit this as is (with linking PRIVATE) and eventually fix the PUBLIC linking in a subsequent commit. I would not have a problem using PUBLIC linking but as you see from the error above I have issues linking with TeleSculptor. 

Another consideration is that I understand you guys work on updating to VTK 9. That might change things, so we could look into fixing the PUBLIC linking then.

Reply:
@dstoup Matt explained this - the problem is that methods from vtkCamera are part of the vtkKwiverCamera but are resolved from the vtk library. So indeed, vtkRenderingCore needs to be public.

Reply:
The error disappears in TeleSculptor if I don't prefix vital_algo_vtk with kwiver::